libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/ws/WebSocketNetworkTransport.kt

```diff
@@ -397,6 +399,10 @@ private constructor(
       this.reopenWhen = reopenWhen
     }
 
+    fun reopenServerUrl(reopenServerUrl: (suspend () -> String)?) = apply {
```

Comment:
What about  `fun serverUrl(serverUrl: (suspend () -> String)) = apply {`? Rely on parameter overloads and fail if someone calls both the `String` and `function` methods?